src/figuresClasses.ts

```diff
@@ -1,19 +1,86 @@
-export interface Figure {
-
+// import { Triangle } from './figuresClasses';
+enum Shape {
+  triangle = 'triangle',
+  circle = 'circle',
+  rectangle = 'rectangle',
 }
 
-export class Triangle {
+enum Color {
+  red = 'red',
+  green = 'green',
+  blue = 'blue',
+}
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  public shape;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = Shape.triangle;
+
+    if (a + b <= c
+      || a + c <= b
+      || b + c <= a
+    ) {
+      throw new Error('sides 1, 2 and 3 can\'t form a triangle');
+    }
+  }
```

Comment:
What about checking if the sides of the triangle are <=0 and throwing the corresponding error?